### ISSUE TITLE
Regarding functionality of sync_batchnorm

### ISSUE THREAD

**Issue by ankitpatnala** [vturrisi/solo-learn#184]
I was wondering what does syncbatch norm do ? 
Isn't it calculates the batch normalization across the data shared among multiple GPUs ?
If that so, don't we leak intra-batch information ? 
![image](https://user-images.githubusercontent.com/26856470/140761328-78fdc962-9f8d-416d-81d7-7a6e1b1a9638.png)
I guess for shuffling BN we shouldn't pass this flag..
https://github.com/vturrisi/solo-learn/blob/45c143dfac6e8a09258081b893726474dba04640/bash_files/pretrain/imagenet-100/mocov2plus.sh#L10

Please correct me if i misunderstood somewhere !

**Comment by vturrisi** [comment]
Hey.
AFAIK, MoCo is the only method that does this shuffling. Our MoCo implementation doesn't do the shuffling and indeed `--sync_batchnorm` avgs across the gpus. We might wanna implement the shuffling just for MoCo and see how it compares.
